### DISCUSSION TITLE
Invalid sampling rate

### DISCUSSION THREAD

**Discussion by georgetasioulis** [akvorado/akvorado#59] [Q&A] [answered]
Hello,

I'm testing out akvorado on a edge Brocade MLX exporing sflow on default port 6343
The sampling rate on the MLX is set to 1024:
```
sflow sample 1024
```
Before running akvorado for the first time (I'm using `docker-compose`) I set `default-sampling-rate: 1024` under the `core` stanza in `akvorado.yaml` and then started it up.
Akvorado however reports the flows as having a 2048 sampling rate. 
```
# curl -s http://127.0.0.1:8081/api/v0/inlet/flows?\limit=1
{
    "TimeReceived": 1659099443,
    "SequenceNum": 293131695,
    "SamplingRate": 2048,
    "ExporterName": "xxxxx",
    "ExporterRole": "edge",
    "ExporterSite": "mlx",
[...]
```
Afterwards I read in the documentation that `default-sampling-rate` should only be set if the device doesn't advertise the sampling rate, so I went ahead and killed the inlet `docker-compose kill akvorado-inlet`, removed the `default-sampling-rate: 1024` line from `akvorado.yaml` and started it again `docker-compose up -d`.

Unfortunately, flows still report a 2048 sampling rate. Also the graphs show half the bandwidth that they really are. So while I'm doing ~1Gbps traffic in reality, the Akvorado graph shows ~500Mbps.

Any ideas?

**Comment by vincentbernat** [comment, marked as answer]
The most plausible explanation is that your device is sending 2048 as a sampling rate. You could confirm that with a packet capture.

**Reply by vincentbernat** [reply]
The latest version of Akvorado now has a `override-sampling-rate` setting to override the sampling rate returned by a router.

**Reply by georgetasioulis** [reply]
After running a packet capture I confirmed with the device was indeed sending 2048 as a sampling rate.
![Screenshot 2022-07-30 at 2 49 47 PM](https://user-images.githubusercontent.com/460954/182162870-a32ad78f-5b18-442a-affe-532666de75f5.png)

Apparently it's the default for Brocade routers. I removed the `sflow sample 1024` line from my config and have been sending with a 2048 sampling rate for the past couple of days:

```
SSH@mlx-4.xxx#show sflow 

sFlow services are enabled.
sFlow management VRF is enabled.
sFlow management VRF name is default-vrf.

sFlow agent IP address: x.x.x.x
sFlow agent IPV6 address: unspecified
sFlow source IP address: x.x.x.x, UDP 8888
sFlow source IPv6 address: unspecified, UDP 8888
Collector IP x.x.x.x, UDP 6343
Polling interval is 20 seconds.
Configured default sampling rate: 1 per 2048 packets.
632965759 UDP packets exported
2990753759 sFlow samples collected.
0 sFlow management-vrf UDP packets dropped
0 ACL sFlow samples collected.
sFlow ports   Global Sample Rate   Port Sample Rate   Hardware Sample Rate
        1/1                 2048               2048   2048
        3/2                 2048               2048   2048
        4/3                 2048               2048   2048
```

Everything appears to be running fine, except that the traffic levels I see in Akvorado do not match my actual ones.

Here's an aggregate of the above three interfaces:
<img width="902" alt="Screenshot 2022-08-01 at 5 20 56 PM" src="https://user-images.githubusercontent.com/460954/182174628-471f8a7e-33c9-4972-9af9-55847ebd3a4f.png">

and per interface graphs:
<img width="820" alt="Screenshot 2022-08-01 at 5 20 14 PM" src="https://user-images.githubusercontent.com/460954/182169485-273fdb14-8033-467d-b38d-2993a0745d8c.png">

And here's what Akvorado shows:
<img width="1061" alt="Screenshot 2022-08-01 at 5 16 15 PM" src="https://user-images.githubusercontent.com/460954/182169762-d0e9ab94-b401-409c-a50a-f5e9b149edf6.png">

per interface:
<img width="1920" alt="Screenshot 2022-08-01 at 5 20 36 PM" src="https://user-images.githubusercontent.com/460954/182169456-6179d3ce-925f-4f7c-bc8c-7e3ae4245ef1.png">

Maybe the 2048 sampling rate is too high for this kind of traffic levels? 
I just switched to 512 and will monitor the results in the next hours, but if you have any suggestions let me know :)

**Reply by vincentbernat** [reply]
Be sure your interface are classified correctly (or remove the InIfBoundary filter). Maybe that's why.
